src/java/org/apache/cassandra/transport/Dispatcher.java

```diff
@@ -46,11 +47,7 @@
 public class Dispatcher
 {
     private static final Logger logger = LoggerFactory.getLogger(Dispatcher.class);
-    
-    private static final LocalAwareExecutorService requestExecutor = SHARED.newExecutor(DatabaseDescriptor.getNativeTransportMaxThreads(),
-                                                                                        DatabaseDescriptor::setNativeTransportMaxThreads,
-                                                                                        "transport",
-                                                                                        "Native-Transport-Requests");
+    private static final LocalAwareExecutorService requestExecutor = Stage.NATIVE_TRANSPORT_REQUESTS.executor();
```

Comment:
Could you please check if using the same executor in 3 places (here, stargate and Stages enum) is safe in terms closing this executor? I.e. is it safe to close the executor in any of them? Should it be closed at all?
kudos @jacek-lewandowski

Reply:
@ethan-brown2022 could you please take a look?